react_juce/core/ShadowView.h

```diff
@@ -160,147 +106,54 @@ namespace reactjuce
-                                                        | YGPrintOptionsChildren));
-#endif
-
-            if (props.contains(layoutAnimatedProp))
-            {
-                if (props[layoutAnimatedProp].isBool() && props[layoutAnimatedProp])
-                {
-                    // Default parameters
-                    return flushViewLayoutAnimated(50.0, 45, BoundsAnimator::EasingType::Linear);
-                }
-
-                if (props[layoutAnimatedProp].isObject())
-                {
-                    double const durationMs = props[layoutAnimatedProp].getProperty(durationProp, 50.0);
-                    double const frameRate = props[layoutAnimatedProp].getProperty(frameRateProp, 45);
-                    int const et = props[layoutAnimatedProp].getProperty(easingProp, 0);
-
-                    return flushViewLayoutAnimated(durationMs, static_cast<int> (frameRate), static_cast<BoundsAnimator::EasingType>(et));
-                }
-            }
-
-            view->setFloatBounds(getCachedLayoutBounds());
-            view->setBounds(getCachedLayoutBounds().toNearestInt());
-
-            for (auto& child : children)
-                child->flushViewLayout();
-        }
+        virtual void flushViewLayout();
 
         /** Recursive traversal of the shadow tree, flushing layout bounds to the
          *  associated view components smoothly over time. This recursive step allows
          *  an animation of a component subtree by just marking the parent as animated.
          */
-        virtual void flushViewLayoutAnimated(double const durationMs, int const frameRate, BoundsAnimator::EasingType const et)
-        {
-            auto viewCurrentBounds = view->getBounds().toFloat();
-            auto viewDestinationBounds = getCachedLayoutBounds();
-
-            animator = std::make_unique<BoundsAnimator>(
-                durationMs,
-                frameRate,
-                et,
-                viewCurrentBounds,
-                viewDestinationBounds,
-                [safeView = juce::Component::SafePointer(view)](auto && stepBounds) {
-                    if (auto* v = safeView.getComponent()) {
-                        v->setFloatBounds(stepBounds);
-                        v->setBounds(stepBounds.toNearestInt());
-                    }
-                }
-            );
-
-            for (auto& child : children) {
-                child->flushViewLayoutAnimated(durationMs, frameRate, et);
-            }
-        }
+        virtual void flushViewLayoutAnimated(double durationMs, int frameRate, BoundsAnimator::EasingType et);
 
     protected:
         //==============================================================================
-        YGNodeRef yogaNode;
-        View* view = nullptr;
-        juce::NamedValueSet props;
+        std::vector<ShadowView*>& getChildren();
 
-        std::unique_ptr<BoundsAnimator> animator;
-        std::vector<ShadowView*> children;
+        class ShadowViewPimpl;
```

Comment:
I think these things can be private, right? Because the only references are within the ShadowView_Yoga.cpp implementation? Or does the TextView need to reference them as well?

Reply:
Yeah this is the one bit I'm not particularly happy with right now. `TextShadowView` needs these at the moment. I think a better thing might emerge but I'm not sure what that is right now with the current shape of `TextShadowView`.